### ISSUE TITLE
GET handler returning 404 interrupts the upload process; ng-flow 2.7.1

### ISSUE THREAD

**Issue by maclav3** [patdek/gongflow#2]
According to https://github.com/flowjs/flow.js#handling-get-or-test-requests, if the GET handler

> returns a permanent error status, upload is stopped

The code 404 that is currently returned by gongflow.ChunkStatus() is recognized as a permanent error code in ng-flow 2.7.1 (flow.js 2.10.1). So if resumable uploads are enabled, the upload stops after the first GET request.

The solution is either to change the permanent error codes list not to include 404 or to intercept the 404 code from ChunkStatus() and return another HTTP code from the handler instead.

**Comment by robertmeta** [comment]
You are 100% right!  Will try to get to it later this week.

**Comment by maclav3** [comment]
Of course, the permanent error codes in ng-flow are configurable, so this is just a minor issue. I mention it mainly because it prevents the normal cooperation of the two libraries on default settings.
